src/convert/closures.rs

```diff
@@ -1,19 +1,38 @@
-#![allow(clippy::fn_to_numeric_cast)]
+#![allow(clippy::fn_to_numeric_cast, coherence_leak_check)]
```

Comment:
Can we keep allowing `coherence_leak_check` as local as possible? We don't want to accidentally add more implementation of that in the future.

Reply:
Literally all the traits in this module need it due to implementations for `Fn(&A)` potentially conflicting with `Fn(A)`, so it seemed to make sense to hoist it out, but sure, no problem with keeping it copy-pasting on each trait either.

Reply:
Indeed, I would prefer keeping it as-is.

Reply:
I moved it to just the reference macro call instead, so we don't need to duplicate it yet it's explicitly allowed only on the one case where it's necessary.